core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1082,6 +1083,7 @@ private[spark] class TaskSetManager(
   }
 
   def recomputeLocality(): Unit = {
+    if (isZombie) return
```

Comment:
A zombie TSM may call this while `executorLost` happens.

Reply:
shall we avoid calling `executorLost` for zombie TSM?

Reply:
We need it because there may be some running tasks on that lost executor for the zombie TSM and the zombie TSM should handle them properly as failed tasks:
https://github.com/apache/spark/blob/e1ea806b3075d279b5f08a29fe4c1ad6d3c4191a/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L938-L948